csrc/preseg_passes/propagate_shardings.cpp

```diff
@@ -271,7 +264,10 @@ void PropagateShardingsPass::runPass(Fusion* fusion) {
       continue;
     }
 
-    int64_t did_pos = selectiveReorderDIDToFront(ref_output, {});
+    std::unordered_set<ParallelType> selected_parallel_types =
+        getParallelTypesToPropagate(sharding_candidates);
+    int64_t did_pos =
+        selectiveReorderDIDToFront(ref_output, selected_parallel_types);
```

Comment:
@wujingyue this is where we use `getParallelTypesToPropagate` in backprop

Reply:
Got it. I was trying to find it in the old code and failed. LGTM!